### PR TITLE
Intermittent QA changes

### PR DESCRIPTION
**Pull request by rfornea**
-adds a bunch of new methods to process_paid_sessions, also adds unit tests.

The process_paid_sessions file currently has a lot of redundancy.  We can clean it up after mainnet.

### CODE REVIEW

**Comment by rfornea** [comment]
I expect some eth tests to potentially fail for this PR.

File: services/eth_gateway.go [hunk 1]
```diff
@@ -552,14 +568,25 @@ func buryPrl(msg OysterCallMsg) bool {
...
-	contractAddress := common.HexToAddress("0xb7baab5cad2d2ebfe75a500c288a4c02b74bc12c")
-	walletKey := getWallet()
+	contractAddress := common.HexToAddress(oysterPearlContract)
+
+	/*TODO: get this keystore stuff working or remove it*/
+	//walletKey := getWallet()
+
+	// TODO:  pull the lines below this out if keystore stuff gets fixed
+	//walletAddress := MainWalletAddress
+	privateKeyString := MainWalletKey
```
[the remark this answers is not in the document]
Comment (rfornea) [reply services/eth_gateway.go]: For the time being I'm not actually sure how we'll be storing MainWalletKey.  Astor has it as a keyStore, but when I added it to the .env I was expecting it to be just the key.  

Using the keystore would be more secure, but I'm having a bit of trouble with the getWallet() method.  I'll sync up with Astor when he starts working and see if we can figure the getWallet method out.  We may end up pulling out all those lines that deal with the keys, if we can resolve the getWallet() issues.  If we end up having to keep them, I'll clean things up.

File: services/eth_gateway.go [hunk 2]
```diff
@@ -642,11 +669,25 @@ func claimPRLs(receiverAddress common.Address, treasureAddress common.Address, t
...
+
+	if privateKeyString[0:2] != "0x" && privateKeyString[0:2] != "0X" {
+		privateKeyString = "0x" + privateKeyString
+	}
```
[the remark this answers is not in the document]
Comment (rfornea) [reply services/eth_gateway.go]: SGTM

File: services/eth_gateway.go [hunk 3]
```diff
@@ -684,13 +725,48 @@ func claimPRLs(receiverAddress common.Address, treasureAddress common.Address, t
...
-	walletKey := getWallet()
+
+	/*TODO: get this keystore stuff working or remove it*/
+	//walletKey := getWallet()
+
+	// TODO:  pull the lines below this out if keystore stuff gets fixed
+	//walletAddress := MainWalletAddress
+	privateKeyString := MainWalletKey
+
+	if privateKeyString[0:2] != "0x" && privateKeyString[0:2] != "0X" {
```
[the remark this answers is not in the document]
Comment (rfornea) [reply services/eth_gateway.go]: Fixed

File: services/eth_gateway.go [hunk 4]
```diff
@@ -684,13 +725,48 @@ func claimPRLs(receiverAddress common.Address, treasureAddress common.Address, t
 */
 func sendPRL(msg OysterCallMsg) bool {
 
+	fmt.Println(msg)
```
[the remark this answers is not in the document]
Comment (rfornea) [reply services/eth_gateway.go]: SGTM

File: models/treasure.go [hunk 5]
```diff
@@ -108,15 +121,39 @@ func (t *Treasure) GetPRLAmount() *big.Int {
...
-		raven.CaptureError(err, nil)
-		return treasureToBury, err
+func GetTreasuresToBuryByPRLStatus(prlStatuses []PRLStatus) ([]Treasure, error) {
+	treasureRowsToReturn := make([]Treasure, 0)
+	for _, prlStatus := range prlStatuses {
+		treasureToBury := []Treasure{}
+		err := DB.RawQuery("SELECT * from treasures where prl_status = ?", prlStatus).All(&treasureToBury)
+		if err != nil {
+			oyster_utils.LogIfError(err)
+			return treasureToBury, err
```
Comment (pzhao5) [inline models/treasure.go]: nit: nil is better here

**Pull request merged**